### PR TITLE
EPROD-424 used evmc client in register evm agent

### PR DESCRIPTION
**Pull request by valeriyr**
# Issue ticket

Issue ticket link: <>

### Code conventions

- [ ] I have performed a self-review of my code
- [ ] Every new function is documented
- [ ] Object names are auto explicative

### Security

- [ ] The PR does not break APIs backward compatibility
- [ ] The PR does not break the stable storage backward compatibility

### Testing

- [ ] Every function is properly unit tested
- [ ] I have added integration tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

File: src/register-evm-agent/src/register_evm_agent/src/reservation.rs [hunk 1]
```diff
@@ -41,72 +46,58 @@ impl<'a> ReservationService<'a> {
...
-            }
-            Err(_) => Err(Error::CouldNotCheckRegistrationStatus(
-                address.to_hex_str(),
-                principal,
-            )),
+
+        if reserved {
+            info!("{address} is already registered");
+        } else {
+            info!("{address} is not registered yet");
```
Comment (ufoscout) [inline src/register-evm-agent/src/register_evm_agent/src/reservation.rs:87]: registered -> reserved

File: src/register-evm-agent/src/register_evm_agent/src/reservation.rs [hunk 2]
```diff
@@ -41,72 +46,58 @@ impl<'a> ReservationService<'a> {
...
-                info!("{address} is not registered yet");
-                Ok(res)
-            }
-            Err(_) => Err(Error::CouldNotCheckRegistrationStatus(
-                address.to_hex_str(),
-                principal,
-            )),
+
+        if reserved {
+            info!("{address} is already registered");
```
Comment (ufoscout) [inline src/register-evm-agent/src/register_evm_agent/src/reservation.rs:85]: registered -> reserved

**Review by ufoscout — APPROVED** [review]

File: src/register-evm-agent/src/register_evm_agent/src/reservation.rs [hunk 3]
```diff
@@ -41,72 +46,58 @@ impl<'a> ReservationService<'a> {
     }
 
     async fn reserve_ic_agent(&self) -> Result<()> {
-        let principal = user_principal(&self.agent)?;
-        info!("registering ic-agent {principal}");
+        info!("registering ic-agent {}", self.agent_principal);
```
Comment (itsyaasir) [inline src/register-evm-agent/src/register_evm_agent/src/reservation.rs:49]:
```suggestion
        info!("reserving ic-agent {}", self.agent_principal);
```

File: src/register-evm-agent/src/register_evm_agent/src/error.rs [hunk 4]
```diff
@@ -16,14 +17,14 @@ pub enum Error {
     Agent(AgentError),
     #[error("wallet is already registered: {0}")]
     AlreadyRegistered(Principal),
```
Comment (itsyaasir) [inline src/register-evm-agent/src/register_evm_agent/src/error.rs:19]:
```suggestion
    AlreadyReserved(Principal),
```

**Review by itsyaasir — APPROVED** [review]

File: src/register-evm-agent/src/register_evm_agent/src/error.rs (hunk 4 above)
Comment (valeriyr) [reply src/register-evm-agent/src/register_evm_agent/src/error.rs]: fixed

File: src/register-evm-agent/src/register_evm_agent/src/reservation.rs (hunk 3 above)
Comment (valeriyr) [reply src/register-evm-agent/src/register_evm_agent/src/reservation.rs]: fixed

File: src/register-evm-agent/src/register_evm_agent/src/reservation.rs (hunk 1 above)
Comment (valeriyr) [reply src/register-evm-agent/src/register_evm_agent/src/reservation.rs]: fixed

File: src/register-evm-agent/src/register_evm_agent/src/reservation.rs (hunk 2 above)
Comment (valeriyr) [reply src/register-evm-agent/src/register_evm_agent/src/reservation.rs]: fixed

**Pull request merged**